addons/test_mail/tests/test_mail_activity.py

```diff
@@ -29,6 +29,79 @@ def setUpClass(cls):
         ])
 
 
+@tests.tagged('mail_activity')
+class TestActivitySearch(TestActivityCommon):
+
+    def test_search_on_date_done_includes_archived(self):
+        """
+        Test that searching on 'date_done' on mail.activity includes archived (done)
+        activities, as implemented in the custom _search method.
+        """
+        active_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Active To-Do',
+            'date_deadline': '2025-09-01',
+        })
+
+        done_activity = self.env['mail.activity'].create({
+            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
+            'summary': 'Done To-Do',
+            'date_deadline': '2025-09-02',
+        })
```

Comment:
All else being equal, this is always better
```suggestion
        active_activity, done_activity = self.env['mail.activity'].create([{
            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
            'res_id': self.test_record.id,
            'user_id': self.user_employee.id,
            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
            'summary': 'Active To-Do',
            'date_deadline': '2025-09-01',
        }, {
            'res_model_id': self.env['ir.model']._get_id('mail.test.activity'),
            'res_id': self.test_record.id,
            'user_id': self.user_employee.id,
            'activity_type_id': self.env.ref('mail.mail_activity_data_todo').id,
            'summary': 'Done To-Do',
            'date_deadline': '2025-09-02',
        }])
```

Could also create the archived record first, then create that activity in batch too, then do
`(done_activity + done_activity_on_archived_record).action_done()`